#*********************************************************
#เรื่องนี้สำคัญมากๆ

# Exception Handling 
# ข้อผิดพลาด การจัดการ
# Exception ข้อผิดพลาดที่เกิดขึ้นตอนโปรเเกรมทำงาน
# try-except-finally เป็นคำสั่งสำหรับ Exception

try :           # คำสั่งที่ใช้ในการดักจับ มีได้เเค่ 1 ตัว
    
    lag1 = int(input('ป้อนตัวเลขตัวที่ 1 : ' ))
    lag2 = int(input('ป้อนตัวเลขตัวที่ 2 : ' ))

    sum = lag1 + lag2
    han = lag1 / lag2

    print(f"ผลรวมของ {lag1} + {lag2} คือ {sum}")
    print(f"ผลรวมของ {lag1} / {lag2} คือ {han}")

except ValueError :
   
    print("ป้อนเเต่ตัวเลขเท่านั้นห้ามตัวอักษร.....สงสัยถามผีในห้อง")

except ZeroDivisionError :
    
    print("ป้อนตัวเลขที่ 2 ห้ามเป็น 0 .....สงสัยถามผีในห้องไอสัส")

except Exception :      #except ครอบจักรวาลสามารถจัดการได้ทุก Exception มีได้หลายตัว
    
    print("มีข้อผิดพลาดเกืดขึ้น ต้องขอโทดนะครับ ต้องการอะไรถามผีในห้องได้เลยครับ")

finally : # สั่งทุกครั้ง จะ Exception หรือไม่ มีได้เเค่ 1 ตัว
    
    print("คิคิ")
    print("หุหุ")
